rpc/lib/server/handlers.go

```diff
@@ -109,35 +109,35 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger log.Logger) http.Han
 		var request types.RPCRequest
 		err := json.Unmarshal(b, &request)
 		if err != nil {
-			WriteRPCResponseHTTPError(w, http.StatusBadRequest, types.NewRPCResponse("", nil, fmt.Sprintf("Error unmarshalling request: %v", err.Error())))
+			WriteRPCResponseHTTP(w, types.RPCParseError("", errors.Wrap(err, "Error unmarshalling request")))
 			return
 		}
```

Comment:
from spec:

```
A Notification is a Request object without an "id" member.
...
 The Server MUST NOT reply to a Notification, including those that are within a batch request.
```

So I think we need to check the request.ID here and not return anything if it's empty

Reply:
fixed